src/goLanguageServer.ts

```diff
@@ -71,6 +75,8 @@ export async function registerLanguageFeatures(ctx: vscode.ExtensionContext) {
 		promptForUpdatingTool(toolName);
 	}
 
+	const latestDiagnostics = new Map<string, vscode.Diagnostic[]>();
```

Comment:
Why not use the `uri` as key for the map and avoid the string conversions when setting and getting the values?

Reply:
The URI that comes into the `didSave` request has a lot more fields set than the one returned by the language server, so they can't be used as map keys. I did change it to use the `path` field instead of `toString()`.